web/onos-gui/src/app/app.component.ts

```diff
@@ -42,20 +56,68 @@ export class AppComponent {
   }
 
   listChanges() {
-    this.diags.requestChanges();
+    this.changes.length = 0;
+    this.diags.requestChanges((change: Change) => {
+      this.changes.push(change);
+    });
   }
 
   listDeviceSummary() {
-    this.admin.requestSummary();
+    this.admin.requestSummary((e: grpcWeb.Error, r: DeviceSummaryResponse) => {
+        if (e != null) {
+          console.warn(e.code, e.message);
+          return;
+        }
+        this.deviceCount = r.getCount()
+    });
   }
 
-  listGnmiCapabilities() {
+  listGnmiCapabilities(): void {
     this.gnmi.requestCapabilities((e: grpcWeb.Error, r: CapabilityResponse) => {
       if (e != null) {
-        console.log('Error getting capabilities from onos-config');
-        console.log(e.code, e.message);
+        console.warn('Error getting capabilities from onos-config');
+        console.warn(e.code, e.message);
       }
       this.capabilites = r;
     });
   }
+
+  getConfigByDevice(deviceId: string): void {
+    this.gnmi.requestGetAllByDevice(deviceId, (e, r) => this.handleGetResponse(e, r));
+  }
+
+  /**
+   * Callback for handling the GetResponse
+   * @param e error
+   * @param r GetResponse
+   */
+  handleGetResponse(e: grpcWeb.Error, r: GetResponse): void {
+    console.log('GetResponse received');
```

Comment:
want to add the response here in the log?

Reply:
will do